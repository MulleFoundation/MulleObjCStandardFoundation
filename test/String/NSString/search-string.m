//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
//#import "MulleStandaloneObjCFoundation.h"


static void  _test( NSString *s, NSString *other, NSUInteger options, char *name)
{
   NSRange   range;
   NSString  *substring;

   range = [s rangeOfString:other
                    options:options];
   substring = @"";
   if( range.length)
      substring = [s substringWithRange:range];

   printf( "%s %s of %s = %s\n",
      [s UTF8String], name, [other UTF8String], [substring UTF8String]);
}


static void  test( NSString *s, NSString *other)
{
   _test( s, other, 0, "default");
   _test( s, other, NSAnchoredSearch, "NSAnchoredSearch");
   _test( s, other, NSBackwardsSearch, "NSBackwardsSearch");
   _test( s, other, NSAnchoredSearch|NSBackwardsSearch, "NSAnchoredSearch|NSBackwardsSearch");
}



int main( int argc, const char * argv[])
{
   // empty string
   test( @"AxB", @"");

   // single letters
   test( @"AxB", @"A");
   test( @"AxB", @"B");
   test( @"xAxBx", @"A");
   test( @"xAxBx", @"B");

   // multiple letters
   test( @"ABCxxx", @"ABC");
   test( @"xxxABC", @"ABC");
   test( @"xxxABCxxx", @"ABC");

   return( 0);
}
