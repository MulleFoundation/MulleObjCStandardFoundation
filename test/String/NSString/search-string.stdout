AxB default of  = 
AxB NSAnchoredSearch of  = 
AxB NSBackwardsSearch of  = 
AxB NSAnchoredSearch|NSBackwardsSearch of  = 
AxB default of A = A
AxB NSAnchoredSearch of A = A
AxB NSBackwardsSearch of A = A
AxB NSAnchoredSearch|NSBackwardsSearch of A = 
AxB default of B = B
AxB NSAnchoredSearch of B = 
AxB NSBackwardsSearch of B = B
AxB NSAnchoredSearch|NSBackwardsSearch of B = B
xAxBx default of A = A
xAxBx NSAnchoredSearch of A = 
xAxBx NSBackwardsSearch of A = A
xAxBx NSAnchoredSearch|NSBackwardsSearch of A = 
xAxBx default of B = B
xAxBx NSAnchoredSearch of B = 
xAxBx NSBackwardsSearch of B = B
xAxBx NSAnchoredSearch|NSBackwardsSearch of B = 
ABCxxx default of ABC = ABC
ABCxxx NSAnchoredSearch of ABC = ABC
ABCxxx NSBackwardsSearch of ABC = ABC
ABCxxx NSAnchoredSearch|NSBackwardsSearch of ABC = 
xxxABC default of ABC = ABC
xxxABC NSAnchoredSearch of ABC = 
xxxABC NSBackwardsSearch of ABC = ABC
xxxABC NSAnchoredSearch|NSBackwardsSearch of ABC = ABC
xxxABCxxx default of ABC = ABC
xxxABCxxx NSAnchoredSearch of ABC = 
xxxABCxxx NSBackwardsSearch of ABC = ABC
xxxABCxxx NSAnchoredSearch|NSBackwardsSearch of ABC = 
